### PR TITLE
Prevent situation update and chain situations together

### LINKED ISSUE #576: Ne pas permettre la modification d'une situation
Une situation enregistrée ne doit pas être modifiable.

En cas de modification dans l'application, il faudrait indiquer l'ID de la situation originale dans l'attribut `previous_id` de la nouvelle situation par exemple.

### PR DESCRIPTION
**Pull request by guillett**
Fixes #576

Helps investigations for issues similar to #589 (cf. https://github.com/sgmap/mes-aides-ui/issues/576#issuecomment-308729307)

### CODE REVIEW

File: app/js/services/situationService.js [hunk 1]
```diff
@@ -120,6 +120,7 @@ angular.module('ddsCommon').factory('SituationService', function($http, $session
             }
 
             var result = {
+                referenceId: situation._id,
```
Comment (MattiSG) [inline app/js/services/situationService.js]: I don't find this name very clear. Pretty much anything could be a “reference”. What about `parent`, `parentSituation`, `previousVersion`, `modifiedFrom`… ?

Comment (guillett) [inline app/js/services/situationService.js]: I used that specific word to reflect OpenFisca naming convention. I agree with your point but given OpenFisca's definition of reference, it makes sense to be consistent.

**Comment by guillett** [comment]
The commit on the API was ready, the PR was not created. I prefer to keep `PUT` support for a few days to prevent errors for current users.

File: app/js/services/situationService.js (hunk 1 above)
Comment (MattiSG) [inline app/js/services/situationService.js]: I'm not sure OpenFisca's “reference” means what you think it means :wink:
See https://github.com/openfisca/openfisca-core/pull/525

**Comment by MattiSG** [comment]
> I prefer to keep PUT support for a few days to prevent errors for current users.

This is scary. Does this mean we don't master our cache invalidation? Could there be any users impacted other than the ones who started a simulation before deployment and finish it after deployment?

**Pull request closed without merging**